pkg/operator/ceph/csi/secrets.go

```diff
@@ -203,5 +279,24 @@ func CreateCSISecrets(context *clusterd.Context, clusterInfo *client.ClusterInfo
 		return errors.Wrap(err, "failed to create kubernetes csi secret")
 	}
 
+	updateCephStatusWithCephxStatus(context, clusterInfo, clusterSpec, clusterNamespaced, shouldRotateCephxKeys)
+
 	return nil
 }
+
+func updateCephStatusWithCephxStatus(context *clusterd.Context, clusterInfo *client.ClusterInfo, clusterSpec *cephv1.ClusterSpec, name types.NamespacedName, shouldRotateCephxKeys bool) {
+	cephCluster := &cephv1.CephCluster{}
+	if err := context.Client.Get(clusterInfo.Context, name, cephCluster); err != nil {
+		if kerrors.IsNotFound(err) {
+			logger.Debugf("CephCluster %q resource not found. Ignoring since object must be deleted", name)
```

Comment:
If the cluster is deleted, then we should kick off a new reconcile to process the deletion so we aren't continuing to provision for a cluster that's no longer present. I think this should return an error so that happens.